net/instaweb/rewriter/rewrite_context.cc

```diff
@@ -2016,10 +2033,24 @@ void RewriteContext::Propagate(bool render_slots) {
     if (has_parent()) {
       parent()->partitions()->mutable_debug_message()->MergeFrom(
           partitions_->debug_message());
-    } else if (render_slots && num_slots() >= 1) {
-      Driver()->InsertDebugComments(partitions_->debug_message(),
-                                    slot(0)->element());
+    } else if (num_slots() >= 1) {
+      if (render_slots) {
+        Driver()->InsertDebugComments(partitions_->debug_message(),
+                                      slot(0)->element());
+      }
+      else if (render_op == RenderOp::kRenderOnlyCspWarning) {
+        StringPiece name = id();
+        RewriteFilter* filter = Driver()->FindFilter(id());
+        if (filter != nullptr) {
+          name = filter->Name();
+        }
+        Driver()->InsertDebugComment(
+            StrCat("PageSpeed output (by ", name, ") not permitted by Content "
```

Comment:
I can't remember; do we need to escape debug comment strings?

Reply:
InsertDebugComment does it.
https://github.com/pagespeed/mod_pagespeed/blob/04e3c5b42e2c7fdacb9dd0deaaeadbff2a3447d4/net/instaweb/rewriter/rewrite_driver.cc#L3235